Yosemite/Yosemite/Stores/CardPresentPaymentStore.swift

```diff
@@ -101,6 +103,19 @@ private extension CardPresentPaymentStore {
         }.store(in: &cancellables)
     }
 
+    func disconnect(onCompletion: @escaping (Result<Void, Error>) -> Void) {
+        cardReaderService.disconnect().sink(receiveCompletion: { error in
+            switch error {
+            case .failure(let error):
+                onCompletion(.failure(error))
+            default:
+                break
+            }
+        }, receiveValue: { (result) in
+            onCompletion(.success(result))
+        }).store(in: &cancellables)
```

Comment:
I'd suggest using a `Subscribers.Sink` subscriber instead of storing the cancellable, like the one in `startCardReaderDiscovery` from #4120. Since this is using a `Future`, it will send a completion right after sending its only value, so we can get automatic cleanup with `Sink`.

Reply:
Good catch, thanks!